peerdid/peer_did.py

```diff
@@ -24,23 +28,60 @@ def create_peer_did_numalgo_0(inception_key: PublicKeyAuthentication) -> PEER_DI
+    :raises ValueError: if the inception_key is not instance base58 encoded
     :return: generated peer_did
     """
-    pass
+    if not isinstance(inception_key, PublicKeyAuthentication):
+        raise TypeError(f"Wrong type of inception_key: {str(type(inception_key))}. Expected: PublicKeyAuthentication")
+    if not _check_if_base58_encoded(inception_key.encoded_value):
+        raise ValueError(f"Inception key is not base58 encoded")
+    peer_did = 'did:peer:0z' + _create_encnumbasis(inception_key)
+    return peer_did
 
 
 def create_peer_did_numalgo_2(encryption_keys: List[PublicKeyAgreement], signing_keys: List[PublicKeyAuthentication],
-                              service_endpoint: str) -> PEER_DID:
+                              services: Union[List[JSON], JSON]) -> PEER_DID:
     """
     Generates peer_did according to the second algorithm
     (https://identity.foundation/peer-did-method-spec/index.html#generation-method)
     For this type of algorithm did_doc can be obtained from peer_did
     :param encryption_keys: list of encryption keys
     :param signing_keys: list of signing keys
-    :param service_endpoint: JSON string conforming to the DID specification (https://www.w3.org/TR/did-core/#services)
+    :param services: List of JSON string conforming to the DID specification (https://www.w3.org/TR/did-core/#services)
+    :raises TypeError:If at least one of encryption keys is not instance of PublicKeyAgreement
+        or at least one of signing keys is not instance of PublicKeyAuthentication
+    :raises ValueError: if at least one of encryption keys is not base58 encoded
+    or at least one of signing keys is not base58 encoded
     :return: generated peer_did
     """
-    pass
+    for key in encryption_keys:
+        if not isinstance(key, PublicKeyAgreement):
+            raise TypeError(f'Wrong type of encryption_key {key}: {str(type(key))}')
+        if not _check_if_base58_encoded(key.encoded_value):
+            raise ValueError(f"Encryption key: {key} is not base58 encoded")
+    for key in signing_keys:
+        if not isinstance(key, PublicKeyAuthentication):
+            raise TypeError(f'Wrong type of signing_key {key}: {str(type(key))}')
+        if not _check_if_base58_encoded(key.encoded_value):
+            raise ValueError(f"Signing key: {key} is not base58 encoded")
+    if not isinstance(services, list):
+        if services == '':
+            services = []
+        else:
+            services = [services]
+    encryption_keys_str = ''
+    if encryption_keys:
+        encryption_keys_str = '.Ez' + '.Ez'.join(_create_encnumbasis(key) for key in encryption_keys)
+    signing_keys_str = ''
+    if signing_keys:
+        signing_keys_str = '.Vz' + '.Vz'.join(_create_encnumbasis(key) for key in signing_keys)
+    services_str = ''
+    if services:
+        services_str = ''.join([_encode_service(service) for service in services])
```

Comment:
That's actually a question whether every service must be appended with .S or we can treat it as just some JSON and append just one .S regardless of number of service elements there. I think the latter is easier and matches the spec more.